Source/Utils/IntervalsParser.H

```diff
@@ -0,0 +1,146 @@
+#ifndef WARPX_INTERVALSPARSER_H_
+#define WARPX_INTERVALSPARSER_H_
+
+/**
+ * \brief This class is a parser for slices of the form i:j:k where i, j and k are integers
+ * representing respectively the starting point, the stopping point and the period.
+ */
+class SliceParser
+{
+public:
+    /**
+    * \brief Constructor of the SliceParser class.
+    *
+    * @param[in] instr an input string of the form "i:j:k", "i:j" or "k" where i, j and k are
+    * integers representing respectively the starting point, the stopping point and the period.
+    * Any of these integers may be omitted in which case it will be equal to their default value
+    * (0 for the starting point, std::numeric_limits<int>::max() for the stopping point and 1 for
+    * the period). For example SliceParser(":1000:") is equivalent to SliceParser("0:1000:1").
+    */
+    SliceParser (const std::string& instr);
+
+    /**
+    * \brief A method that returns true if the input integer is contained in the slice. (e.g. if
+    * the list is initialized with "300:500:100", this method returns true if and only if n is
+    * 300, 400 or 500). If the period is negative or 0, the function always returns false.
+    *
+    * @param[in] n the input integer
+    */
+    bool contains (const int n) const;
+
+    /**
+    * \brief A method that returns the smallest integer strictly greater than n such that
+    * contains(n) is true. Returns std::numeric_limits<int>::max() if there is no such integer.
+    *
+    * @param[in] n the input integer
+    */
+    int nextContains (const int n) const;
+
+    /**
+    * \brief A method that returns the greatest integer strictly smaller than n such that
+    * contains(n) is true. Returns 0 if there is no such integer.
+    *
+    * @param[in] n the input integer
+    */
+    int previousContains (const int n) const;
+
+    /**
+    * \brief A method that returns the slice period.
+    *
+    */
+    int getPeriod () const;
+
+    /**
+    * \brief A method that returns the slice start.
+    *
+    */
+    int getStart () const;
+
+    /**
+    * \brief A method that returns the slice stop.
+    *
+    */
+    int getStop () const;
+
+private:
+    int m_start = 0;
+    int m_stop = std::numeric_limits<int>::max();
```

Comment:
Maybe I miss something... is this modified at any point? If not, could we use `0` and `std::numeric_limits<int>::max()` instead?

Reply:
Yes (if I understood your question correctly), `m_start` and `m_stop` are modified in the constructor of this class and correspond to the 2 first integers in an input string of the form `500:1000:100` (when we do something like `amrex::is_it<int>(insplit[0], m_start)` this assigns the value in `insplit[0]` to `m_start`; this is not very visible, in particular since it's inside an assert).